src/extract/Extract.ml

```diff
@@ -2785,13 +2791,16 @@ let extract_trait_decl_coq_arguments (ctx : extraction_ctx) (fmt : F.formatter)
     decl.parent_clauses;
   (* The required methods *)
   List.iter
-    (fun (item_name, id) ->
+    (fun (item_name, bound_fn) ->
+      let open Charon.Types in
+      let fun_id = bound_fn.binder_value.fun_id in
       (* Lookup the definition to retrieve the information about the explicit/implicit parameters *)
-      let trans = A.FunDeclId.Map.find id ctx.trans_funs in
+      let trans = A.FunDeclId.Map.find fun_id ctx.trans_funs in
       let f = trans.f in
       let explicit_info =
         explicit_info_drop_prefix decl.generics f.signature.explicit_info
       in
+      (* TODO: this is likely incorrect, we should instantiate the binders properly *)
```

Comment:
@sonmarcho This will likely cause issues after https://github.com/AeneasVerif/charon/issues/127: the full method generics will no longer be a simple concatenation of the impl generics + method generics (because I'll be adding the extra associated type parameters at the end of the list of parameters instead of in the middle). We should find a way to instantiate the binders properly here.